spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -432,6 +449,15 @@ else if (payload instanceof String) {
 					throw new IllegalArgumentException(
 							"unsupported Message payload type [" + payload.getClass().getName() + "]");
 				}
+				if (this.preserveTimestamp) {
+					if (timestamp instanceof Number) {
+						resultFile.setLastModified(((Number) timestamp).longValue());
+					}
+					else {
+						logger.warn("Could not set lastModified, header " + FileHeaders.SET_MODIFIED
+								+ " must be a Number, not " + timestamp.getClass());
```

Comment:
Non-constant log messages should be wrapped with the `if (this.logger.is...Enabled()) {`